activestorage/lib/active_storage/service/url_config.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+module ActiveStorage
+  class Service::UrlConfig # :nodoc:
+    def initialize(uri, configuration_hash = {})
+      @uri = uri
+      @configuration_hash = configuration_hash.merge(build_url_hash).freeze
+    end
+
+    def fetch(_)
+      @configuration_hash.fetch(:url)
+    end
+
+    private
+      attr_reader :uri
+
+      def build_url_hash
+        service = uri.scheme && uri.scheme.tr("-", "_")
```

Comment:
This can be refactored:

```ruby
uri.scheme&.tr("-", "_")

```

Reply:
This is just code I copied from AR:
https://github.com/rails/rails/blob/e3da4fc53d5da7be1d5b93d5659e77c8daddccea/activerecord/lib/active_record/database_configurations/connection_url_resolver.rb#L83